Fixed BlockMiner run when sync is off and miner=start

zilm13 opened the pull request:
#596 not an issue for develop, but when sync is off and miner=start EthereumJ fails to start. This PR fixes it.

Nashatyrev commented:
Just merged conflicting PR https://github.com/ethereum/ethereumj/pull/590
Can you please merge your fix again?

zilm13 commented:
Please note than:
1) I've added pruneManager setting in StandaloneBlockchain
2) I've fixed miner tests, plus MineBlock and ExternalMinerTest are now with mocks
